crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -342,6 +342,37 @@ impl C8YHttpProxy for JwtAuthHttpProxy {
     }
 }
 
+pub struct FakeC8YHttpProxy {}
```

Comment:
This is an interesting one, since the http_proxy has now been moved out to a separate crate just marking it with `cfg(test)` doesn't work anymore, I have moved all of the tests and the struct inside tests module.